src/main/java/org/openrewrite/java/migrate/jakarta/HasNoJakartaNullAnnotations.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.jakarta;
+
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.ScanningRecipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.search.FindAnnotations;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.marker.SearchResult;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class HasNoJakartaNullAnnotations extends ScanningRecipe<AtomicBoolean> {
+    @Override
+    public String getDisplayName() {
+        return "Project has no Jakarta null annotations";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Search for @Nonnull and @Nullable annotations, mark all source as found if no annotations are found.";
+    }
+
+    @Override
+    public AtomicBoolean getInitialValue(ExecutionContext ctx) {
+        return new AtomicBoolean();
+    }
```

Comment:
Would this be another case where we'd want ``Map<JavaProject, Boolean>` or even `Set<JavaProject>` to account for multi module projects? Now it's a boolean across all modules, whereas I think we'd want to evaluate per JavaProject.

Reply:
Yes you are right. Let me reimplement it exactly like [ModuleHasPlugin](https://github.com/openrewrite/rewrite/blob/facbcfb3da944b3b08dc6bab41c5d5a478254c29/rewrite-maven/src/main/java/org/openrewrite/maven/search/ModuleHasPlugin.java)! As it's nearly the same recipe, but with a different condition.